api/src/main/java/org/openmrs/api/impl/UserServiceImpl.java

```diff
@@ -80,14 +80,9 @@ public User createUser(User user, String password) throws APIException {
 		
 		checkPrivileges(user);
 		
-		// if a password wasn't supplied, throw an error
-		if (password == null || password.length() < 1) {
```

Comment:
I agree with you, also stumbled on an AOP call leading to unreachable code :(
but I think this has to be addressed in a separate ticket, please create it 👍 . this ticket is as far as I know only about adding tests, I therefore suggest only to touch the UserServiceTest